Library/Formula/llvm.rb

```diff
@@ -116,18 +151,74 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    if build.with? "clang"
+      (buildpath/"tools/clang").install resource("clang")
     end
 
-    if build.with? "clang"
+    if build.with? "extra-tools"
+      if build.with? "clang"
+        (buildpath/"tools/clang/tools/extra").install resource("extra-tools")
+      else
+        odie "clang's extra tools must be built with clang"
+      end
+    end
+
+    if build.with? "libcxx"
       (buildpath/"projects/libcxx").install resource("libcxx")
-      (buildpath/"tools/clang").install resource("clang")
-      (buildpath/"tools/clang/tools/extra").install resource("clang-tools-extra")
+      (buildpath/"projects/libcxxabi").install resource("libcxxabi")
+    end
+
+    if build.with? "lld"
+      (buildpath/"tools/lld").install resource("lld")
+    end
+
+    if build.with? "lldb"
+      if build.with? "clang"
+        (buildpath/"tools/lldb").install resource("lldb")
+
+        # Building lldb requires a code signing certificate.
+        # The instructions provided by llvm creates this certificate in the
+        # user's login keychain. Unfortunately, the login keychain is not in
+        # the search path in a superenv build. The following three lines add
+        # the login keychain to ~/Library/Preferences/com.apple.security.plist,
+        # which adds it to the superenv keychain search path.
+        mkdir_p "#{ENV["HOME"]}/Library/Preferences"
+        username = ENV["USER"]
+        system "security", "list-keychains", "-d", "user", "-s", "/Users/#{username}/Library/Keychains/login.keychain"
+      else
+        odie "lldb must be built along with clang"
+      end
+    end
+
+    if build.with? "polly"
+      if build.with? "clang"
+        (buildpath/"tools/polly").install resource("polly")
+      else
+        odie "polly must be built along with clang"
+      end
+    end
+
+    if build.with? "sanitizers"
+      if build.with? "clang"
+        (buildpath/"projects/compiler-rt").install resource("sanitizers")
+
+        # compiler-rt has some iOS simulator features that require i386 symbols
+        # I'm assuming the rest of clang needs support too for 32-bit compilation
+        # to work correctly, but if not, perhaps universal binaries could be
+        # limited to compiler-rt. llvm makes this somewhat easier because compiler-rt
+        # can almost be treated as an entirely different build from llvm.
+        ENV.permit_arch_flags
+
+      else
+        odie "sanitizers must be built along with clang"
+      end
     end
 
-    (buildpath/"tools/lld").install resource("lld") if build.with? "lld"
-    (buildpath/"tools/lldb").install resource("lldb") if build.with? "lldb"
+    if build.with?("tests") &&
+       build.with?("sanitizers") &&
+       build.without?("libcxx")
```

Comment:
I think I had it earlier, but IIRC I took it out since it failed every time I tested it. Looks like I forgot to take the rest out when I decided not to include tests.